File: 351004/Levental/task350-project/discussion/src/main/java/by/bsuir/task330/discussion/dto/NoticeRequestTo.java
package by.bsuir.task330.discussion.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record NoticeRequestTo(
        Long id,
        @NotNull Long articleId,
        @NotBlank @Size(min = 2, max = 2048) String content
) {}

File: 351003/brezgunov/discussion/src/main/kotlin/com/example/discussion/dto/notice/NoticeRequestTo.kt
package com.example.discussion.dto.notice

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.util.UUID

data class NoticeRequestTo (
    val id: UUID? = null,

    @field:NotNull
    val country: String,

    @field:NotNull
    val newsId: Long,

    @field:NotBlank
    @field:Size(min = 4, max = 2048)
    val content: String
)